ckan/controllers/group.py

```diff
@@ -104,22 +99,25 @@ def _render_template(self, template_name):
         ''' render the correct group/org template '''
         return render(self._replace_group_org(template_name))
 
-    def _redirect_to(self, *args, **kw):
-        ''' wrapper to ensue the correct controller is used '''
-        if self.group_type == 'organization' and 'controller' in kw:
-            kw['controller'] = 'organization'
+    def _redirect_to__this_controller(self, *args, **kw):
+        ''' wrapper around redirect_to but it adds in this request's controller
+        (so that it works for Organization or other derived controllers)'''
+        kw['controller'] = request.environ['pylons.routes_dict']['controller']
```

Comment:
redirects and url_for should keep the same controller as the current request. That is better than trying to determine it from the group_type, which may be a custom one.